File: app/src/main/java/kr/hs/emirim/ohm/goout_dialog.java
package kr.hs.emirim.ohm;

import android.app.Dialog;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;

public class goout_dialog extends AppCompatActivity {

    private Button OK_button;
    private Button cancel_button;

    Dialog dilaog03; //다이얼로그2

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_goout_dialog);

        OK_button = (Button) findViewById(R.id.end_button);
        cancel_button = (Button) findViewById(R.id.cancel_button);

        dilaog03 = new Dialog(goout_dialog.this); //다이얼로그 초기화
        dilaog03.requestWindowFeature(Window.FEATURE_NO_TITLE); //타이틀 제거
        dilaog03.setContentView(R.layout.activity_goout_dialog); //레이아웃 연결

        OK_button.setOnClickListener(new View.OnClickListener() { // 나가기 창
            @Override
            public void onClick(View v) {
                showDialog03();
            }

            public void showDialog03() {
                dilaog03.show();

                Button okBtn = dilaog03.findViewById(R.id.ok_button);
                okBtn.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        Intent intent = new Intent(goout_dialog.this, HomeActivity.class);
                        startActivity(intent);
                        dilaog03.dismiss(); // 다이얼로그 닫기
                    }
                });
            }
        });

        cancel_button.setOnClickListener(new View.OnClickListener() { // 나가기 창
            @Override
            public void onClick(View v) {
                showDialog03();
            }

            public void showDialog03() {
                dilaog03.show();

                Button CancelBtn = dilaog03.findViewById(R.id.cancel_button);
                CancelBtn.setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        dilaog03.dismiss(); // 다이얼로그 닫기
                    }
                });
            }
        });


    }
}

File: app/src/main/java/kr/hs/emirim/ohm/goout_dialog.kt
package kr.hs.emirim.ohm

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle

class goout_dialog : AppCompatActivity() {

    var dilaog03 //다이얼로그
    private lateinit var ok_button : Button //확인버튼
    private lateinit var cancel_button : Button //취소버튼

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_goout_dialog)

        ok_button = findViewById(R.id.ok_button)
        cancel_button = findViewById(R.id.cancel_button)

        dilaog03 = android.app.Dialog(this@goout_dialog) //다이얼로그 초기화
        dilaog03.requestWindowFeature(android.view.Window.FEATURE_NO_TITLE) //타이틀 제거
        dilaog03.setContentView(R.layout.activity_goout_dialog) //레이아웃 연결

        // 나가기 창
        open fun onClick(v: android.view.View?) {
            showDialog03()
        }

        open fun showDialog03() {
            dilaog03.show()
            val endBtn: android.widget.Button? =
                dilaog03.findViewById<android.widget.Button?>(R.id.ok_button)
            okBtn.setOnClickListener(object : android.view.View.OnClickListener {
                open fun onClick(view: android.view.View?) {
                    val intent: Intent? = Intent(this@goout_dialog, HomeActivity::class.java)
                    startActivity(intent)
                    dilaog03.dismiss() // 다이얼로그 닫기
                }
            })
            dilaog03.findViewById<android.view.View?>(R.id.cancel_button)
                .setOnClickListener(object : android.view.View.OnClickListener {
                    open fun onClick(view: android.view.View?) {
                        val intent: Intent? = Intent(this@goout_dialog, ChatingActivity::class.java)
                        startActivity(intent)
                        dilaog03.dismiss() // 다이얼로그 닫기
                    }
                })
        }
    }
}